custom/ucr_ext/tests.py

```diff
@@ -0,0 +1,382 @@
+import uuid
+from xml.etree import ElementTree
+from datetime import date, datetime
+from django.test import SimpleTestCase, TestCase
+from corehq.apps.userreports.expressions.factory import ExpressionFactory
+from corehq.apps.userreports.specs import EvaluationContext
+from corehq.apps.receiverwrapper.util import submit_form_locally
+from casexml.apps.case.mock import CaseStructure, CaseFactory
+from casexml.apps.case.tests.util import delete_all_cases, delete_all_xforms
+
+
+def _safe_text(input_value):
+    if input_value is None:
+        return ''
+    try:
+        return str(input_value)
+    except:

```

Comment:
nit: should probably be `except Exception` to avoid trapping unexpected runtime things.